tapir/wirgarten/views/form_product_type.py

```diff
@@ -0,0 +1,165 @@
+        )
+
+
+def create_product_type(form: ProductTypeForm, period_id):
+    # product type
+    print("\t[create] ProductType ")
+    pt = ProductType.objects.create(
+        name=form["name"],
+        pickup_enabled=form["pickup_enabled"],
+    )
+    # tax rate
+    TaxRate.objects.create(
+        product_type_id=pt.id,
+        tax_rate=form["tax_rate"],
+        valid_from=date.today(),
+        valid_to=None,
+    )
+    # capacity
+    ProductCapacity.objects.create(
+        period_id=period_id,
+        product_type=pt,
+        capacity=form["capacity"],
+    )
+
+
+@require_http_methods(["GET", "POST"])
+def get_product_type_edit_form(request, **kwargs):
+    print("-- kwargs: ", kwargs)
+    # if this is a POST request we need to process the form data
+    if request.method == "POST":
+        # create a form instance and populate it with data from the request:
+        form = ProductTypeForm(request.POST)
+        # check whether it's valid:
+        if form.is_valid():
+            # process the data in form.cleaned_data as required
+            save_product_type(form.cleaned_data)
+            # redirect to a new URL:
+            return HttpResponseRedirect(
+                reverse_lazy(PAGE_ROOT) + "?" + request.environ["QUERY_STRING"]
+            )
+
+    # if a GET (or any other method) we'll create a blank form
+    else:
+        form = ProductTypeForm(**kwargs)
+
+    return render(
+        request, "wirgarten/product_cfg/period_product_cfg.html", {"form": form}
+    )
+
+
+@require_http_methods(["GET", "POST"])
+def get_product_type_add_form(request, **kwargs):
+    # if this is a POST request we need to process the form data
+    if request.method == "POST":
+        # create a form instance and populate it with data from the request:
+        form = ProductTypeForm(request.POST)
+        # check whether it's valid:
+        if form.is_valid():
+            # process the data in form.cleaned_data as required
+            form.cleaned_data["period_id"] = kwargs["periodId"]
+            create_product_type(form.cleaned_data, kwargs["periodId"])
+            # redirect to a new URL:
+            return HttpResponseRedirect(
+                reverse_lazy(PAGE_ROOT) + "?" + request.environ["QUERY_STRING"]
+            )
+
+    # if a GET (or any other method) we'll create a blank form
+    else:
+        form = ProductTypeForm(**kwargs)
+
+    return render(
+        request, "wirgarten/product_cfg/period_product_cfg.html", {"form": form}
+    )
+
+
+@require_http_methods(["GET"])
+def delete_product_type(request, **kwargs):
+    try:
+        ProductCapacity.objects.get(
+            period__id=kwargs["periodId"], product_type__id=kwargs["prodTypeId"]
+        ).delete()
+
+        try:
+            if ProductCapacity.objects.get(product_type__id=kwargs["prodTypeId"]):
+                print(
+                    "Delete ProductType: Product type is still referenced, just deleting product capacity."
```

Comment:
not sure about that.. but we will create a seperate issue for the business logic